substrate/client/src/lib.rs

```diff
@@ -176,7 +181,7 @@ impl<B, E> Client<B, E> where
 			blockchain::BlockStatus::Unknown => return Ok(ImportResult::UnknownParent),
 		}
 
-		let mut transaction = self.backend.begin_transaction(BlockId::Number(header.number))?;
+		let mut transaction = self.backend.begin_transaction(BlockId::Number(header.number - 1))?;
```

Comment:
why `- 1`?

Reply:
The argument to `begin_transaction` is the base (parent) block id. Either hash or number.

Reply:
Changed it to `parent_hash` anyway